### PR TITLE
[Server] Designs: add share functionality

### PR DESCRIPTION
**Pull request by MUzairS15**
Signed-off-by: MUzairS15 <muzair.shaikh810@gmail.com>

**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by theBeginner86 — DISMISSED** [review]
LGTM

File: server/handlers/user_handler.go [hunk 1]
```diff
@@ -120,3 +120,14 @@ func (h *Handler) UserPrefsHandler(w http.ResponseWriter, req *http.Request, pre
...
+
+func (h *Handler) ShareDesignHandler(w http.ResponseWriter, r *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+	statusCode, err := provider.ShareDesign(r)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error: %v", err.Error()), statusCode)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	fmt.Fprint(w, "Design shared successfully")
```
Comment (leecalcote) [inline server/handlers/user_handler.go:132]: No more “successfully”s, please. 
```suggestion
	fmt.Fprint(w, "Design shared")
```

File: server/models/error.go [hunk 2]
```diff
@@ -338,3 +339,7 @@ func ErrSavingSeededComponents(err error, content string) error {
 func ErrDownloadingSeededComponents(err error, content string) error {
 	return errors.New(ErrDownloadingSeededComponentsCode, errors.Alert, []string{"Could not download seed content for" + content}, []string{err.Error()}, []string{"The content is not present at the specified url endpoint", "HTTP requests failed"}, []string{"Make sure the content is available at the endpoints", "Make sure that Github is reachable and the http requests are not failing"})
 }
+
+func ErrShareDesign(err error) error {
+	return errors.New(ErrShareDesignCode, errors.Alert, []string{"cannot make design shareable"}, []string{err.Error()}, []string{"email id provided might not be valid", "insufficient permission"}, []string{"try sharing designs of which you are the owner", "Retry after sometime", "try using alternate email id"})
```
Comment (leecalcote) [inline server/models/error.go:344]:
```suggestion
	return errors.New(ErrShareDesignCode, errors.Alert, []string{"cannot make design public"}, []string{err.Error()}, []string{"email address provided might not be valid", "insufficient permission"}, []string{"Ensure that you are the owner of the design you are sharing", "Try again later", "Try using an alternate email address"})
```

File: server/router/server.go [hunk 3]
```diff
@@ -198,6 +198,8 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int, g http.
 		Methods("GET")
 	gMux.Handle("/api/application/{id}", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.DeleteMesheryApplicationHandler), models.ProviderAuth))).
 		Methods("DELETE")
+	gMux.Handle("/api/user/share", h.ProviderMiddleware((h.AuthMiddleware(h.SessionInjectorMiddleware(h.ShareDesignHandler), models.ProviderAuth)))).
```
Comment (leecalcote) [inline server/router/server.go:201]: Let’s move this out of the /user area and into /content

Comment (leecalcote) [reply server/router/server.go:201]: I have changed this API location.

And have run `make swagger-docs-build`.

**Comment by l5io** [comment]
🚀 Preview for commit bb7bd30cb4aa34ea18910bd2f0749007a8e576d3 at: https://63d99439378f37006f959085--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a5b3ff1ec5b3cc7f9736450d5e7edfe7d140f1c3 at: https://63da38567c52a2741db6716b--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a6d24f8a1a994f5273e54a02897270c892373583 at: https://63da4f9f5f0d800637044df3--meshery-docs-preview.netlify.app

**Review by leecalcote — APPROVED** [review]

**Pull request merged**